Makefile

```diff
@@ -69,55 +78,70 @@ publish: build lint
 
 .PHONY: clean
 clean: ## Remove intermediate build artifacts & cache.
-	@rm -rf $(BUILDDIR)/*
+	@rm -rf $(BUILDDIR)
 	@rm -rf vendor
+	@rm $(VERSIONFILE)
 
 .PHONY: test
 test: ## Run unit tests.
+test: buildbox
 	docker run $(DOCKERFLAGS) \
 		--env="GO111MODULE=off" \
 		$(BUILDBOX) \
 		dumb-init go test -cover -race -v ./infra/... ./lib/config/...
 
 .PHONY: lint
 lint: ## Run static analysis against source code.
-lint: buildbox
+lint: vendor buildbox
 	docker run $(DOCKERFLAGS) \
 		--env="GO111MODULE=off" \
-		$(BUILDBOX) dumb-init golangci-lint run \
-		--skip-dirs=vendor \
-		--timeout=2m
+		$(BUILDBOX) dumb-init golangci-lint run
+
+.PHONY: vendor
+vendor: ## Download dependencies into vendor directory.
+vendor: vendor/modules.txt
+
+vendor/modules.txt: go.mod | $(VERSIONFILE) $(BUILDBOX_IIDFILE)
+	docker run $(DOCKERFLAGS) $(BUILDBOX) \
+		dumb-init go mod vendor
+	@touch vendor/modules.txt
 
 .PHONY: version
 version: ## Show the robotest version.
 version: $(VERSIONFILE)
 	@echo "Robotest Version: $(VERSION)"
 
-VERSION_GO="/* DO NOT EDIT THIS FILE. IT IS GENERATED BY make */\n\n\
-package robotest\n\
-const(\n\
-	Version = \"$(VERSION)\"\n\
-	GitCommit = \"$(GIT_COMMIT)\"\n\
-)\n"
+define VERSION_GO
+/* DO NOT EDIT THIS FILE. IT IS GENERATED BY MAKE */
+
+package robotest
 
-# $(VERSIONFILE) is PHONY because I haven't found a concise & understandable
-# way to tell if the commit has changed or there is a new tag. Unfortunately
-# this does mean it will spuriously retrigger downstream targets -- 2020-04 walt
+const(
+	Version   = "$(VERSION)"
+	GitCommit = "$(GIT_COMMIT)"
+)
+endef
+
+# $(VERSIONFILE) is PHONY because make doesn't know if the `git describe` output
+# has changed (new commit or tag).  Thus we generate a new draft every time & compare
+# to see if the version changed.
 .PHONY: $(VERSIONFILE)
-$(VERSIONFILE): Makefile
-	@printf $(VERSION_GO) | gofmt > $(VERSIONFILE)
-	@echo "version metadata saved to $(VERSIONFILE)"
+$(VERSIONFILE): export CONTENT=$(VERSION_GO)
+$(VERSIONFILE): TMP = $(BUILDDIR)/$(VERSIONFILE).tmp
+$(VERSIONFILE):
+	@echo "$$CONTENT" > $(TMP)
+	@if ! cmp -s $(TMP) $(VERSIONFILE); then \
+		echo "$$CONTENT" > $(VERSIONFILE); \
+        echo "version metadata saved to $(VERSIONFILE)"; \
+    fi
```

Comment:
nit: indent

Reply:
Fixed in 713ebfab6dbf641155b7ef022a7b4529d2dd35cd.